plugins/outputs/azuremonitor/azuremonitor.go

```diff
@@ -0,0 +1,445 @@
+		log.Printf("W! attempted to aggregate metric over 30 minutes old: %v, %v", t, tbucket)
+		return
+	}
+
+	// Azure Monitor doesn't have a string value type, so convert string
+	// fields to dimensions (a.k.a. tags) if enabled.
+	if a.StringAsDimension {
+		for fk, fv := range m.Fields() {
+			if v, ok := fv.(string); ok {
+				m.AddTag(fk, v)
+			}
+		}
+	}
+
+	for _, f := range m.FieldList() {
+		fv, ok := convert(f.Value)
+		if !ok {
+			continue
+		}
+
+		// Azure Monitor does not support fields so the field
+		// name is appended to the metric name.
+		name := m.Name() + "_" + sanitize(f.Key)
+		id := hashIDWithField(m.HashID(), f.Key)
+
+		_, ok = a.cache[tbucket]
+		if !ok {
+			// Time bucket does not exist and needs to be created.
+			a.cache[tbucket] = make(map[uint64]*aggregate)
+		}
+
+		nf := make(map[string]interface{}, 4)
+		nf["min"] = fv
+		nf["max"] = fv
+		nf["sum"] = fv
+		nf["count"] = 1
+		// Fetch existing aggregate
+		agg, ok := a.cache[tbucket][id]
+		if ok {
+			aggfields := agg.Fields()
+			if fv > aggfields["min"].(float64) {
+				nf["min"] = aggfields["min"]
+			}
+			if fv < aggfields["max"].(float64) {
+				nf["max"] = aggfields["max"]
+			}
+			nf["sum"] = fv + aggfields["sum"].(float64)
+			nf["count"] = aggfields["count"].(int64) + 1
+		}
+
+		na, _ := metric.New(name, m.Tags(), nf, tbucket)
+		a.cache[tbucket][id] = &aggregate{na, true}
+	}
+}
+
+func convert(in interface{}) (float64, bool) {
+	switch v := in.(type) {
+	case int64:
+		return float64(v), true
+	case uint64:
+		return float64(v), true
+	case float64:
+		return v, true
+	case bool:
+		if v {
+			return 1, true
+		}
+		return 0, true
+	default:
+		return 0, false
+	}
+}
+
+var invalidNameCharRE = regexp.MustCompile(`[^a-zA-Z0-9_]`)
+
+func sanitize(value string) string {
+	return invalidNameCharRE.ReplaceAllString(value, "_")
+}
+
+func hashIDWithField(id uint64, fk string) uint64 {
+	h := fnv.New64a()
+	b := make([]byte, binary.MaxVarintLen64)
+	n := binary.PutUvarint(b, id)
+	h.Write(b[:n])
+	h.Write([]byte("\n"))
+	h.Write([]byte(fk))
+	h.Write([]byte("\n"))
+	return h.Sum64()
+}
+
+// Push sends metrics to the output metric buffer
+func (a *AzureMonitor) Push() []telegraf.Metric {
+	var metrics []telegraf.Metric
+	for tbucket, aggs := range a.cache {
+		// Do not send metrics early
+		if tbucket.After(time.Now().Add(-time.Minute)) {
+			continue
+		}
+		for _, agg := range aggs {
+			// Only send aggregates that have had an update since
+			// the last push.
+			if !agg.updated {
+				continue
+			}
+			metrics = append(metrics, agg.Metric)
+		}
+	}
+	return metrics
+}
+
+// Reset clears the cache of aggregate metrics
+func (a *AzureMonitor) Reset() {
+	for tbucket := range a.cache {
+		// Remove aggregates older than 30 minutes
+		if tbucket.Before(time.Now().Add(-time.Minute * 30)) {
+			delete(a.cache, tbucket)
+			continue
+		}
+		for id := range a.cache[tbucket] {
+			a.cache[tbucket][id].updated = false
```

Comment:
If the interval is lower than the 1m aggregation period, this can get set to false before the metric has had a chance to be returned in `Push()` since you may have multiple calls to `Reset()` before the metric is returned.  This causes the plugin not to write any metrics unless you set the set the flush_interval to at least `1m`.